### ISSUE TITLE
Why HackRF seems to send datas but smartphone receive nothing (or for a few seconds only)

### ISSUE THREAD

**Issue by xxxxxaaaaaaaannnn** [osqzss/gps-sdr-sim#240]
Hello,

I will explain you here why you see your HackRF sending data and your phone doesn't pickup anything or in more common cases, your **GPS Test** app detect them and after a few seconds they are all gone (_no fix_).

I tested this on Android only.

TL;DR : 
> Android has an anti spoof GPS mechanism and you may want to go into airplane mode, which will **really** disable BlueTooth/Wi-Fi/GSM localisation and thus will trust your GPS faked signals anyway.

Long story : 
My guess is that Android embed a new feature that allows you to unlock your phone when you are _at home_ without using any pin code or pattern. Since this feature exists, they need to make sure that the phone isn't receiving fake GPS signals to get a free unlock. Even if your phone doesn't have the feature, it might be included in the Android build you have.

**Even if your phone is set in mode "GPS only, don't use Wi-Fi"**, your phone will still use Wi-Fi/Bluetooth/GSM to compare with your GPS location, if the difference is too important, GPS will be discarded.

You can try two thing to asses if you are suffering from that : 
1) the easiest is (after configuring your phone in **GPS only** mode in your settings), put your phone in **Airplane mode**, start your hackrf_transmit and run **GPS Test**. It should work :)
2) You can spoof your GPS not too far from where you are really, but I've noticed that 100 meters is already too much ; I guess only tip #1 works.

You also need to make sure you ran gps-sdr-sim with the latest **brdc** and with the **-b 8** argument, this is required for HackRF.

About the transmission power of **hackrf_transfer**, this doesn't really matter, I just tuned the power  (**-x <power>**) to be received on my phone but not too much to annoy my neighbors.

Now if you ask the question on how to use it _without Airplane mode_ because you want to cheat on an application or game, I have no clue. Depending on your phone, you may start the game with Airplane mode enable and then disable it after the game is running. It works for me but not on my friend's phone (new generation).

Good luck !

**Comment by xxxxxaaaaaaaannnn** [comment]
I also found that if you add parameter **-C -4** to hackrf_transmit, it works much better, but I don't know why.

**Comment by eried** [comment]
Thanks for the clarification. I have tried with 3 devices, none gets fix :/ they start losing satelites in view and then lose the fix, not sure why.

What parameters do you use for -x ?

**Comment by petem001** [comment]
The hackrf need a tcxo to keep it on the good frequency and simulate the frequency drift of the sats.

Télécharger Outlook pour Android<https://aka.ms/ghei36>

________________________________

**Comment by eried** [comment]
I have a TCXO, an external "0.5" one.  Downloaded the latest brdc for today (brdc1180.20n), is there is anything else to test?

**Comment by michalsternadel** [comment]
Did you try generating fake signal with 8 bit I/Q data format? [link](https://github.com/osqzss/gps-sdr-sim/issues/245#issuecomment-623588974)

**Comment by eried** [comment]
> Did you try generating fake signal with 8 bit I/Q data format? [link](https://github.com/osqzss/gps-sdr-sim/issues/245#issuecomment-623588974)

Yes, at the end I had some success :) using SatGen free nmea generator https://github.com/eried/portapack-havoc/wiki/GPS-Sim#example-1

After a while 2 of my 4 devices point to that fake position :D

**Comment by whysoft** [comment]
![无标题](https://user-images.githubusercontent.com/26561262/144743068-b304f0b2-3616-480d-b5f6-577c72328136.png)
The   coordinate  cannot be spoofed on Xiaomi phone.
It can be seen from the internal app that there is no  GPGGA message.
I don't know how to create a gpgga signal, but I think if someone can think about and create some signals for gpgga, this may be solved.
